web/srv/api_handlers.go

```diff
@@ -439,18 +439,26 @@ func (h *handler) handleGetExtension(w http.ResponseWriter, req *http.Request, _
 	extensionName := req.FormValue("extension_name")
 
 	resp := map[string]interface{}{}
-	ns, err := h.k8sAPI.GetNamespaceWithExtensionLabel(ctx, extensionName)
-	if err != nil && strings.HasPrefix(err.Error(), "could not find") {
-		renderJSON(w, resp)
-		return
-	} else if err != nil {
-		renderJSONError(w, err, http.StatusInternalServerError)
-		return
+	if extensionName == "" {
+		extensions, err := h.k8sAPI.GetAllNamespacesWithExtensionLabel(ctx)
```

Comment:
It looks like when we get this list of extensions from the k8s API, we send all the API info to the frontend. 
![image](https://user-images.githubusercontent.com/2197104/123662391-08d6f380-d7fb-11eb-8aad-e647991a5605.png)

The only info we need in the frontend is the extension name and the namespace. I would extract that from what we receive from the k8s API and send back a list of objects that contain the extension's name and its namespace.